### PR TITLE
Enhanced logging [Issue #3107]

### PR DESCRIPTION
**Pull request by marcstern**
In case too many keys, indicate this in the audit log

### CODE REVIEW

File: apache2/msc_json.c [hunk 1]
```diff
@@ -374,9 +375,12 @@ int json_process_chunk(modsec_rec *msr, const char *buf, unsigned int size, char
 	if (msr->json->depth_limit_exceeded) {
            *error_msg = "JSON depth limit exceeded";
 	} else {
-           char *yajl_err = yajl_get_error(msr->json->handle, 0, buf, size);
-           *error_msg = apr_pstrdup(msr->mp, yajl_err);
-           yajl_free_error(msr->json->handle, yajl_err);
+        if (msr->json->yajl_error) *error_msg = msr->json->yajl_error;
```
Comment (airween) [inline apache2/msc_json.c:378]: Could you set the correct indentation for this block? Just for the better readability.

**Review by airween — CHANGES_REQUESTED** [review]

**Comment by airween** [comment]
I see that you added a new "extra" line:
```
ModSecurity: JSON parsing error: More than 10 JSON keys [hostname "localhost"] [uri "/post"] [unique_id "ZkUWkhdCjBcNkeH57qvdrgAAAAA"]
ModSecurity: Warning. Match of "eq 0" against "REQBODY_ERROR" required. [file "/etc/modsecurity/modsecurity.conf"] [line "77"] [id "20
0002"] [msg "Failed to parse request body."] [data "JSON parsing error: More than 10 JSON keys"] [severity "CRITICAL"] [hostname "localhost"] [uri "/post"] [unique_id "ZkUWkhdCjBcNkeH57qvdrgAAAAA"]
```

The first line contains the cause and the second the triggered rule details (with the cause too).

It's good for me, just asking: do we need the first line? (it does not contain extra information)

Cc @kkrupka - is this what you would like to have?

**Comment by kkrupka** [comment]
Perfect. Thanks!

**Comment by marcstern** [comment]
I didn't add any line, it's automatically added.
Should be the case before the change (except that it would be the generic message about the "callback").

**Review by airween — APPROVED** [review]

**Comment by airween** [comment]
> I didn't add any line, it's automatically added. Should be the case before the change (except that it would be the generic message about the "callback").

Ok, thanks.

I'm going to merge it now - thank you. Then it closes #3107.

**Pull request merged**